offset_manager.go

```diff
@@ -250,8 +251,12 @@ func (pom *partitionOffsetManager) fetchInitialOffset(retries int) error {
 
 	switch block.Err {
 	case ErrNoError:
-		pom.offset = block.Offset
-		pom.metadata = block.Metadata
+		if block.Offset >= 0 {
+			pom.offset = block.Offset
+			pom.metadata = block.Metadata
+		} else {
+			pom.offset = pom.parent.conf.Consumer.Offsets.Initial
```

Comment:
Right. I would prefer to have this `if` in the actual `Offset()` method I think though; then we can put the `+1` in as well and all the processing logic ends up in one place. Thoughts?